src/Libraries/DynamoWatch3D/Watch3D.cs

```diff
@@ -283,35 +254,24 @@ protected override void DeserializeCore(XmlElement nodeElement, SaveContext cont
             base.DeserializeCore(nodeElement, context);
             try
             {
-                foreach (XmlNode node in nodeElement.ChildNodes)
+                foreach (var node in nodeElement.ChildNodes.Cast<XmlNode>().Where(node => node.Name == "view"))
                 {
-                    if (node.Name == "view")
-                    {
-                        WatchWidth = Convert.ToDouble(node.Attributes["width"].Value);
-                        WatchHeight = Convert.ToDouble(node.Attributes["height"].Value);
-
-                        foreach (XmlNode inNode in node.ChildNodes)
-                        {
-                            if (inNode.Name == "camera")
-                            {
-                                var x = Convert.ToDouble(inNode.Attributes["pos_x"].Value, CultureInfo.InvariantCulture);
-                                var y = Convert.ToDouble(inNode.Attributes["pos_y"].Value, CultureInfo.InvariantCulture);
-                                var z = Convert.ToDouble(inNode.Attributes["pos_z"].Value, CultureInfo.InvariantCulture);
-                                var lx = Convert.ToDouble(inNode.Attributes["look_x"].Value, CultureInfo.InvariantCulture);
-                                var ly = Convert.ToDouble(inNode.Attributes["look_y"].Value, CultureInfo.InvariantCulture);
-                                var lz = Convert.ToDouble(inNode.Attributes["look_z"].Value, CultureInfo.InvariantCulture);
-                                CameraPosition = new Point3D(x, y, z);
-                                LookDirection = new Vector3D(lx, ly, lz);
-                            }
-                        }
-                    }
-                }
+                    if (node.Attributes == null || node.Attributes.Count == 0) continue;
+
+                    WatchWidth = Convert.ToDouble(node.Attributes["width"].Value);
+                    WatchHeight = Convert.ToDouble(node.Attributes["height"].Value);
 
+                    // The deserialization logic is unified between the view model and this node model.
+                    // For the node model, we need to supply the deserialization method with the camera node.
+                    var cameraElement = nodeElement.ChildNodes.Cast<XmlNode>().FirstOrDefault(innerNode => innerNode.Name == "Camera");
+                    initialCameraData = HelixWatch3DViewModel.DeserializeCamera(cameraElement);
+                }
             }
             catch (Exception ex)
             {
                 Log(LogMessage.Error(ex));
                 Log("View attributes could not be read from the file.");
+                initialCameraData = new CameraData();
```

Comment:
Is it ok to move initialCameraData = new CameraData() to the start, instead of doing it in Exception?